File: modules/web/app/controllers/web/WebAdmin.java
package controllers.web;

import javax.inject.Inject;

import controllers.core.AExerciseAdminController;
import model.WebExercise;
import model.WebExerciseReader;
import model.user.User;
import play.data.FormFactory;
import play.mvc.Result;
import play.twirl.api.Html;

public class WebAdmin extends AExerciseAdminController<WebExercise> {

  @Inject
  public WebAdmin(FormFactory theFactory) {
    super(theFactory, WebRoutesObject$.MODULE$, WebExercise.finder, WebExerciseReader.getInstance());
  }

  @Override
  public Result adminIndex() {
    return ok(views.html.webAdmin.index.render(getUser()));
  }

  public Result exRest(int exerciseId) {
    return ok(views.html.webAdmin.webExRest.render(getUser(), finder.byId(exerciseId)));
  }

  @Override
  protected Html renderExEditForm(User user, WebExercise exercise, boolean isCreation) {
    return views.html.webAdmin.editExForm.render(user, exercise);
  }

}

File: modules/web/app/controllers/web/WebAdmin.scala
package controllers.web

import javax.inject.Inject

import controllers.core.AExerciseAdminController
import model.{WebExercise, WebExerciseReader}
import play.data.FormFactory
import play.mvc.{Result, Results}
import play.twirl.api.Html

class WebAdmin @Inject()(f: FormFactory)
  extends AExerciseAdminController[WebExercise](f, WebToolObject, WebExercise.finder, WebExerciseReader) {

  override def statistics = new Html(
    s"""<li>Es existieren insgesamt ${WebExercise.finder.all.size} Aufgaben, davon
       |  <ul>
       |    <li>${WebExercise.withHtmlPart} Aufgaben mit HTML-Teil</li>
       |    <li>${WebExercise.withJsPart} Aufgaben mit JS-Teil</li>
       |  </ul>
       |</li>""".stripMargin)

  def exRest(exerciseId: Int): Result = Results.ok(views.html.webAdmin.webExRest.render(getUser, finder.byId(exerciseId)))

}
